bindata/network/ovn-kubernetes/006-ovs-node.yaml

```diff
@@ -80,10 +80,60 @@ spec:
                 exit 0
             }
             trap quit SIGTERM
+{{ if .EnableIPsec }}
+            # Every time we restart the container, we will create a new key pair
+            echo "Configuring IPsec keys"
+
+            # We use the system-id as the CN. This is a requirement for OVN
+            cn=$(ovs-vsctl get Open_vSwitch . external-ids:system-id | tr -d "\"")
+
+            mkdir -p /etc/openvswitch/keys
+            pushd /etc/openvswitch/keys
+            umask 077 && openssl genrsa -out ipsec-privkey.pem 2048
+            openssl req -new -text \
+                    -extensions v3_req \
+                    -addext "subjectAltName = DNS:${cn}" \
+                    -subj "/C=US/O=ovnkubernetes/OU=kind/CN=${cn}" \
+                    -key ipsec-privkey.pem \
+                    -out ipsec-req.pem
+
+            csr_64=$(cat ipsec-req.pem | base64 | tr -d "\n")
+
+            # The signer controller does not allow re-signing a key. We will
+            # delete the old key to be sure it is not there
+            kubectl delete --ignore-not-found=true csr/$(hostname)
+
+            cat <<EOF | kubectl apply -f -
+            apiVersion: certificates.k8s.io/v1
+            kind: CertificateSigningRequest
+            metadata:
+              name: $(hostname)
+            spec:
+              groups:
+              - system:authenticated
+              request: ${csr_64}
+              signerName: network.openshift.io/signer
+              usages:
+              - client auth
+          EOF
+
+            kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' | base64 -d | openssl x509 -outform pem -text -out ipsec-cert.pem
```

Comment:
Presumably we need to wait for this to be signed, right?

Reply:
Yes, good point.